static_frame/core/index_hierarchy.py

```diff
@@ -474,17 +488,24 @@ def _from_arrays(cls: tp.Type[IH],
                 index_constructors_iter=index_constructors_iter,
                 )
 
+        # NOTE: some index_constructors will change the dtype of the final array
+        if index_constructors is None:
+            blocks = TypeBlocks.from_blocks(arrays)
+            own_blocks = True
+        else:
+            blocks = None
+            own_blocks = True
+
         if name is None:
             name = cls._build_name_from_indices(indices)
 
-        blocks = TypeBlocks.from_blocks(arrays)
 
         return cls(
                 indices=indices,
                 indexers=indexers,
                 name=name,
                 blocks=blocks,
-                own_blocks=True,
+                own_blocks=own_blocks,
```

Comment:
`own_blocks` is always `True` per the above branch

Reply:
Good catch; it should be False on 497!

Reply:
@flexatone Did you fix this? line 497 still says `own_blocks=True`?

Reply:
Thanks for reminding me, I am surprised it did not result in an error!